app/src/main/java/in/avimarine/boatangels/activities/SplashActivity.java

```diff
@@ -0,0 +1,53 @@
+package in.avimarine.boatangels.activities;
+
+import android.app.Dialog;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import in.avimarine.boatangels.R;
+import in.avimarine.boatangels.general.DialogUtils;
+import in.avimarine.boatangels.general.Versioning;
+
+/**
+ * This file is part of an
+ * Avi Marine Innovations project: BoatAngels
+ * first created by aayaffe on 11/10/2018.
+ */
+public class SplashActivity extends AppCompatActivity {
+
+  private static final String TAG = "SplashActivity";
+  @Override
+  protected void onCreate(@Nullable Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    Versioning versioning = new Versioning(this);
+    versioning.getSupportedVersion(result -> {
+      if (versioning.getInstalledVersion()<result)
+      {
+        alertOnUnsupportedVersion();
+      } else {
+        Intent intent = new Intent(this, MainActivity.class);
+        startActivity(intent);
+        finish();
+      }
+    });
+  }
+
+  private void alertOnUnsupportedVersion() {
+    Log.d(TAG, "Version not supported, starting dialog");
```

Comment:
I would add a printing of current installed version number and of the supported one.

Reply:
will add debug printing. 
I do not think this needs to be exposed to the lamen user.